data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/AbstractSqlRepositoryOperations.java

```diff
@@ -523,4 +595,64 @@ public int hashCode() {
     protected interface StatementSupplier<PS> {
         PS create(String ps) throws Exception;
     }
+
+
+    /**
+     * The provider for {@link JsonColumnReader} when JSON columns are being read.
+     *
+     * @param <RS> the result set type
+     */
+    protected static final class JsonColumnReaderProvider<RS> {
+
+        private final JsonColumnReader<RS> defaultJsonColumnReader;
```

Comment:
Extract this class as a bean so we don't need to pass multiple values in the argument.